libcontainer/configs/config.go

```diff
@@ -227,11 +267,32 @@ type Capabilities struct {
 	Ambient []string
 }
 
+func (hooks *Hooks) RunHooks(name HookName, spec *specs.State) error {
```

Comment:
Argument name should be `state` not `spec`.

Reply:
Also, this `RunHooks` helper is associated with the wrong structure IMHO. You could just make `type HookList []Hook` and then define a `Run` method for that. Then you could do

```go
if err := hooks.Prestart.Run(state); err != nil {
    // ...
}
```

No need for `config.HookName` everywhere.

Reply:
Resolved, thanks!